src/main.spec.js

```diff
@@ -47,10 +61,26 @@ test("Checks HTMLElements", () => {
   for (const child of ul.children) {
     expect(child).toBeInstanceOf(HTMLLIElement);
   }
-  expect(ul.children[0].textContent).toBe("Dir");
-  expect(ul.children[1].textContent).toBe("FinalDir");
-  expect(ul.children[2].textContent).toBe("AnotherFile.html");
-  expect(ul.children[3].textContent).toBe("SFile.sth");
-  expect(ul.children[4].textContent).toBe("test.go");
-  expect(ul.children[5].textContent).toBe("Troll.go");
+  expect(ul.children[0].children[0].textContent).toBe("Dir");
+  expect(ul.children[1].children[0].textContent).toBe("FinalDir");
+  expect(ul.children[2].children[0].textContent).toBe("AnotherFile.html");
+  expect(ul.children[3].children[0].textContent).toBe("SFile.sth");
+  expect(ul.children[4].children[0].textContent).toBe("test.go");
+  expect(ul.children[5].children[0].textContent).toBe("Troll.go");
+  for (i = 0; i < ul.length; i++) {
```

Comment:
First, we should not use `ul.length`, as it is not in the spec of the `HTMLUListElement` ([read the spec](https://developer.mozilla.org/en-US/docs/Web/API/HTMLUListElement)).

Next, we should simplify this. An example that could work:

```javascript
for (const li of ul.children) {
  // ...do thing here
}
```